src/main/java/com/linkedin/kmf/services/ClusterTopicManipulationService.java

```diff
@@ -0,0 +1,242 @@
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.CreateTopicsResult;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.requests.DescribeLogDirsResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Service monitoring the creations and deletions of Kafka Cluster's Topic.
+ */
+public class ClusterTopicManipulationService implements Service {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ClusterTopicManipulationService.class);
+  private final String _configDefinedServiceName;
+  private final int _reportIntervalSecond;
+  private final ScheduledExecutorService _executor;
+  private final AdminClient _adminClient;
+  private volatile boolean _isOngoingTopicCreationDone;
+  private String _currentlyOngoingTopic;
+  private CreateTopicsResult _createTopicsResult;
+  private AtomicInteger _totalPartitions = new AtomicInteger();
+  // TODO -- ClusterTopicManipulationMetrics implementation in progress!
+  private ClusterTopicManipulationMetrics _clusterTopicManipulationMetrics;
+
+  public ClusterTopicManipulationService(String name, AdminClient adminClient) {
+    LOGGER.info("ClusterTopicManipulationService constructor initiated {}", this.getClass().getName());
+
+    _isOngoingTopicCreationDone = true;
+    _adminClient = adminClient;
+    _executor = Executors.newSingleThreadScheduledExecutor();
+    _reportIntervalSecond = 1;
+    _configDefinedServiceName = name;
+    // TODO: instantiate a new instance of ClusterTopicManipulationMetrics(..) here.
+  }
+
+  /**
+   * The start logic must only execute once.  If an error occurs then the implementer of this class must assume that
+   * stop() will be called to clean up.  This method must be thread safe and must assume that stop() may be called
+   * concurrently. This can happen if the monitoring application's life cycle is being managed by a container.  Start
+   * will only be called once.
+   */
+  @Override
+  public void start() {
+    LOGGER.info("ClusterTopicManipulationService started for {} - {}", _configDefinedServiceName,
+        this.getClass().getCanonicalName());
+    Runnable clusterTopicManipulationServiceRunnable = new ClusterTopicManipulationServiceRunnable();
+
+    _executor.scheduleAtFixedRate(clusterTopicManipulationServiceRunnable, _reportIntervalSecond, _reportIntervalSecond,
+        TimeUnit.SECONDS);
+  }
+
+  private class ClusterTopicManipulationServiceRunnable implements Runnable {
```

Comment:
I think you can just make `ClusterTopicManipulationService` a `Runnable` and in the start() method, submit `this` to the executor. No need to create another private inner class and the only purpose is to schedule a periodical invocation to the outer class's method

Reply:
let's leave it as a private class for clarity and also for leaving a room in case we will need to add additional functionalities to this Runnable in the future.